pkg/platform/packet/packet_test.go

```diff
@@ -0,0 +1,169 @@
+package packet
+
+import "testing"
+
+func TestCheckNotEmptyWorkers(t *testing.T) {
+	// Test with empty workers, it should fail.
+	c := config{}
+	hclErr := c.checkNotEmptyWorkers()
+
+	// Exactly 1 failure expected: no worker pools.
+	if len(hclErr) != 1 { // //nolint:gomnd
+		t.Errorf("Expected to detect 1 failure, but detected %v failures", len(hclErr))
+	}
+
+	// Test with one worker pool, it should work.
+	c = config{WorkerPools: []workerPool{{Name: "test"}}}
+	hclErr = c.checkNotEmptyWorkers()
+
+	if hclErr != nil {
+		t.Errorf("Validation failed and shouldn't fail")
+	}
+}
+
+func TestCheckWorkerPoolNamesUniqueDup(t *testing.T) {
+	c := config{
+		WorkerPools: []workerPool{
+			{
+				Name: "dup",
+			},
+			{
+				Name: "dup",
+			},
+		},
+	}
+
+	hclErr := c.checkWorkerPoolNamesUnique()
+
+	// Only one failure expected: the duplicated worker pool.
+	if len(hclErr) != 1 { //nolint:gomnd
+		t.Errorf("Expected to detect 1 failure, but detected %v failures", len(hclErr))
+	}
+}
+
+func TestCheckWorkerPoolNamesUniqueNotDup(t *testing.T) {
+	c := config{
+		WorkerPools: []workerPool{
+			{
+				Name: "not",
+			},
+			{
+				Name: "dup",
+			},
+		},
+	}
+
+	hclErr := c.checkWorkerPoolNamesUnique()
```

Comment:
Why not consolidate into single line? `if err := c.checkWorkerPoolNamesUnique(); err != nil {`

Also, if we test for diagnostics, we should use `err.HasErrors()`.

Reply:
Good points!

I guess the first one is because I didn't write go code in a while :-(. Didn't know about the second one, but makes total sense and will check it out in case there other useful helpers!

Reply:
The downside of this is that you can't easily use the len of the slice, though.

I've moved it to use `HasErrors()` and lost precision because I only check if "HasErrors()", so checking that exactly the number of errors expected is returned (this will catch when an error is reported two times for the same line and we don't want it, for example) is not possible.

I'm not sure this is a big loss, though, but... I'm unsure if it's worth it.

Using the len of the slice is wrong, as it is possible that not all entries are errors. We can easily check the number of errors with a function like:

```
func ErrorsNr(d hcl.Diagnostics) int {
      ret := 0

      for _, diag := range d {
              if diag.Severity == hcl.DiagError {
                      ret++
              }
      }

      return ret
}
```

But I've had a quick look at terraform code and the test also use `HasErrors()`. So, I'm leaning towards just using that...

What do you think?

Reply:
Yeah, I think we should use `HasErrors()` and this should be sufficient.